### DISCUSSION TITLE
Execution of script triggered by map save

### DISCUSSION THREAD

**Discussion by doctomlug** [freeplane/freeplane#1390] [General]
Is there a simple (and/or already implemented) way to run a script if the current map gets saved (before it actually gets saved)?
Similar to an objects destructor

This would allow to alter the map programmatically or perform map dependent actions before it gets written to the file system 

Potential use cases:
- possibility to add nodes by scripts (like a tool I am currently playing with) but remove them before saving, thus avoid clutter 
- export map in different format
- upload map to server
- .... 

An idea to trigger it might be something like // @ExecutionModes ({ON_MAP_SAVE})

Thank you

**Comment by euu2021** [comment]
You can create a script to be assigned to "Ctrl+S", so you get your own "custom" way of saving. In that script, you would put all those actions that you are interest, and then finish (or start) with a map save. 

Take a look, for example, at this script, which is a "custom save" that exports as PDF on every save: https://github.com/freeplane/freeplane/discussions/958#discussioncomment-4805302

**Comment by doctomlug** [comment]
Thank you for the idea. 
I should have thought of that by myself.
